src/SimpleWebRequest.ts

```diff
@@ -344,14 +373,14 @@ export abstract class SimpleWebRequestBase<TOptions extends WebRequestOptions =
         _.forEach(nextHeaders, (val, key) => {
             const headerLower = key.toLowerCase();
             if (headerLower === 'content-type') {
-                assert.ok(false, 'Don\'t set Content-Type with options.headers -- use it with the options.contentType property');
+                this._assertAndClean(false, 'Don\'t set Content-Type with options.headers -- use it with the options.contentType property');
                 return;
             }
             if (headerLower === 'accept') {
-                assert.ok(false, 'Don\'t set Accept with options.headers -- use it with the options.acceptType property');
+                this._assertAndClean(false, 'Don\'t set Accept with options.headers -- use it with the options.acceptType property');
                 return;
             }
-            assert.ok(!headersCheck[headerLower], 'Setting duplicate header key: ' + headersCheck[headerLower] + ' and ' + key);
+            this._assertAndClean(!headersCheck[headerLower], 'Setting duplicate header key: ' + headersCheck[headerLower] + ' and ' + key);
```

Comment:
If any of these clean then you should prevent the `xhr` from sending as well. [Note: These `return` are inside `_.forEach`]

Reply:
The assert prevents that from happening - I want to leave the library in a good state if somebody decides to catch the exception.
If they do -> subsequent requests work
If they don't -> there are no subsequent requests
Returning here only makes sense if someone strips/replaces the asserts - in which case they should be responsible for making sure the library still works